lister/lib/lister.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Lister do
+  @moduledoc """
+  Documentation for `Lister`.
+
+  """
+
+  defprotocol Listable do
+    @spec show(any()) :: any()
+    def show(term)
+  end
+
+  defimpl Listable, for: [String, BitString] do
+    @spec show(binary()) :: list()
+    def show(term) do
+      term |> String.to_charlist()
+    end
+  end
+
+  defimpl Listable, for: List do
+    @spec show(list()) :: list()
+    def show(term) do
+      term
+    end
+  end
+
+  defimpl Listable, for: [Float, Integer] do
+    @spec show(integer()) :: list(integer() | float())
+    def show(term) do
+      [term]
+    end
+  end
+
+  defimpl Listable, for: Map do
+    @spec show(map()) :: list(any())
+    def show(term) do
+      term |> Map.to_list()
+    end
+  end
+
+  defimpl Listable, for: Tuple do
+    @spec show(tuple()) :: list(any())
+    def show(term) do
+      term |> Tuple.to_list()
```

Comment:
Technically, one might use `@for` inside an implementation like
```suggestion
      term |> @for.to_list()
```